morphir/runtime/src/org/finos/morphir/runtime/quick/Loop.scala

```diff
@@ -271,9 +270,9 @@ private[morphir] case class Loop(globals: GlobalDefs) extends InvokeableEvaluato
   }
 
   def handleLetRecursion(
-      va: ValueAttribs,
-      definitions: Map[Name, Definition[TypeAttribs, ValueAttribs]],
-      inValue: RuntimeValue,
+      va: UType,
+      definitions: Map[Name, Definition[scala.Unit, UType]],
```

Comment:
So this probably should have been `RuntimeDefinition` before, and should be `TypedDefinition` now.

BTW @Bouzomgi , this is what I briefly mentioned about none of these being enforced because none of the types are opaque - since the compiler can see that these all boil down to `scala.Unit` and `Type[scala.Unit]`, it wasn't enforcing anything about how we aliased or referenced them.